x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/schemas/create_rules_schema.ts

```diff
@@ -67,5 +68,6 @@ export const createRulesSchema = Joi.object({
   type: type.required(),
   threat: threat.default([]),
   references: references.default([]),
+  note: note.allow('').default(''),
```

Comment:
For note, you can do the default of an empty string for all new data pushes, I think that makes sense. However, since you have it setup to be able to accept a "nullable" you could also not set a default and just allow the user to never push a note and then the backend will have a null value for note (which it does already since we haven't had this feature).

Up to you on this one, but if you allow the default to be an empty string for all new data then down the road you will not be able to distinguish between when a user has added a note and it is just empty vs. the user has never added a note since even if the user does not add a note it will end up being an empty string always.

Just pointing it out. For arrays it's pretty simple to default them to empty arrays as that makes sense. An empty array means nothing was added. For strings it gets a bit more tricky. Usually I let the strings be null to distinguish between a note was added vs an empty note.

Up to you though. I am fine either way, just pointing this one thing out.

Reply:
Spoke offline, but just wanted to follow up here. Thanks for bringing those points up. I've updated the code and tests to reflect that `note` will not be defaulted to any value. There wasn't any value added to doing so.